src/archemist/stations/tecan_xlp6000_syringe_pump_station/state.py

```diff
@@ -0,0 +1,95 @@
+        self._model = station_model
+
+    @classmethod
+    def from_dict(cls, station_dict: Dict, liquids: List[Liquid], solids: List[Solid]):
+        model = SyringePumpStationModel()
+        cls._set_model_common_fields(station_dict,model)
+        model._module = cls.__module__
+        model.save()
+        return cls(model)
+
+    @property
+    def status(self) -> SyringePumpStatus:
+        self._model.reload('machine_status')
+        return self._model.machine_status
+
+    @status.setter
+    def status(self, new_status: SyringePumpStatus):
+        self._model.update(machine_status=new_status)
+    
+    #to check if the station is idle 
+
+    def assign_station_op(self, station_op: Any):
+        if isinstance(station_op, SyringePumpDispenseOpDescriptor):
+            self.status = SyringePumpStatus.DISPENSE
+        elif isinstance(station_op, SyringePumpWithdrawOpDescriptor):
+            self.status = SyringePumpStatus.WITHDRAW
+        return super().assign_station_op(station_op)
+    
+    def complete_assigned_station_op(self, success: bool, **kwargs):
+        current_op = self.get_assigned_station_op()
+        if isinstance(current_op, (SyringePumpDispenseOpDescriptor,SyringePumpWithdrawOpDescriptor)) and current_op.was_successful:
+            self.status = SyringePumpStatus.JOB_COMPLETE
+        super().complete_assigned_station_op(success, **kwargs)
+
+''' ==== Station Operation Descriptors ==== '''
+class SyringePumpDispenseOpDescriptor(StationOpDescriptor):
+    def __init__(self, op_model: SyringePumpOpDescriptorModel) -> None:
+        self._model = op_model
+
+    @classmethod
+    def from_args(cls, **kwargs):
+        model = SyringePumpOpDescriptorModel()
+        model.pump_info = kwargs['pump_info']
+        model._type = cls.__name__
+        model._module = cls.__module__
+        return cls(model)
+
+    @property
+    def dispense_port(self) -> int:
+        return self._model.pump_info.get('port')
+
+    @property
+    def dispense_speed(self) -> float:
+        return self._model.pump_info.get('speed')
+
+    @property
+    def dispense_volume(self) -> float:
+        return self._model.pump_info.get('volume')
+
+class SyringePumpWithdrawOpDescriptor(StationOpDescriptor):
+    def __init__(self, op_model: SyringePumpOpDescriptorModel) -> None:
+        self._model = op_model
+
+    @classmethod
+    def from_args(cls, **kwargs):
+        model = SyringePumpOpDescriptorModel()
+        model.pump_info = kwargs['pump_info']
+        model._type = cls.__name__
+        model._module = cls.__module__
+        return cls(model)
+
+    @property
+    def withdraw_port(self) -> int:
+        return self._model.pump_info.get('port')
+        
+
+    @property
+    def withdraw_speed(self) -> float:
+        return self._model.pump_info.get('speed')
+
+    @property
+    def withdraw_volume(self) -> float:
+        return self._model.pump_info.get('volume')
```

Comment:
This op will need to be removed since we are delegating all the low level functionality to the roslabware